dotnet/src/SemanticKernel.Core/Data/TextSearch/VectorStoreTextSearch.cs

```diff
@@ -103,17 +104,16 @@ public VectorStoreTextSearch(
     /// <param name="options">Options used to construct an instance of <see cref="VectorStoreTextSearch{TRecord}"/></param>
     public VectorStoreTextSearch(
         IVectorizableTextSearch<TRecord> vectorizableTextSearch,
-        ITextSearchStringMapper stringMapper,
-        ITextSearchResultMapper resultMapper,
+        ITextSearchStringMapper? stringMapper = null,
+        ITextSearchResultMapper? resultMapper = null,
         VectorStoreTextSearchOptions? options = null)
     {
         Verify.NotNull(vectorizableTextSearch);
-        Verify.NotNull(stringMapper);
-        Verify.NotNull(resultMapper);
 
         this._vectorizableTextSearch = vectorizableTextSearch;
-        this._stringMapper = stringMapper;
-        this._resultMapper = resultMapper;
+        this._propertyReader = new TextSearchResultPropertyReader(typeof(TRecord));
```

Comment:
It would be good to avoid doing reflection unless we need to do it.  E.g. if someone passed their own mapper, we should preferably not do the reflection.  Reflection adds overhead and complicates the AOT story, so it's good to give people the chance to avoid it.